### DISCUSSION TITLE
How can I use Context Menu Component but with left click instead of right click?

### DISCUSSION THREAD

**Discussion by rahul3103** [radix-ui/primitives#2726] [Help]
I wanted to use ContextMenu Component but i want it to work with the left click instead of right click but I couldn't find in the doc anywhere how to change the click.

**Comment by benoitgrelard** [comment]
Hi @rahul3103 I am confused, `ContextMenu` works with right-click, not left-click, the same as native context menus.

**Reply by rahul3103** [reply]
I want a similar kind of feature to open menu on left click

**Reply by benoitgrelard** [reply]
I am still confused, a menu that opens on right-click is exactly what `ContextMenu` is for: https://www.radix-ui.com/primitives/docs/components/context-menu

**Reply by rahul3103** [reply]
https://codesandbox.io/p/devbox/fxkynl
I want to open menu on left click at co-ordinates x and y
I tried with menu component but collision will not work

**Reply by benoitgrelard** [reply]
You are using the wrong component, use `ContextMenu` instead of `DropdownMenu` as I linked above.

**Reply by rahul3103** [reply]
Sorry @benoitgrelard for confusion. I wanted to open menu on left click instead of right click. ContextMenu component supports right click.

**Reply by benoitgrelard** [reply]
That's what `DropdownMenu` is for, but you need a trigger, otherwise it wouldn't be accessible. I'm afraid you are trying to build something that isn't accessible. What is your use-case?

**Reply by rahul3103** [reply]
So, I am using Echarts to render chart and i want to open a menu at x,y co-ordinate which is returned by Echart when user clicks on chart, and for that I was trying to open menu component. So, my use case is like if user click on any X axis component open menu component with different menus , if user clicks on bar than open menu with different component but on (x,y) coordinates where user has clicked.

**Reply by benoitgrelard** [reply]
Yeah I'm not sure about the accessibility of interactive charts really. This isn't something we support as part of our patterns.

**Reply by rahul3103** [reply]
Any guidance, how can I achieve it something similar if possible?

**Reply by benoitgrelard** [reply]
I am not familiar with Echarts. What kind of info are you trying to show when clicking on a bar?

**Reply by rahul3103** [reply]
onClick of suppose bar in barchart, I will show menu such as add formatting, sorting, changing color and other functionalities. I want to open menu when only certain criteria justifies. So, I tried that in this example
https://codesandbox.io/p/devbox/fxkynl
